support/c/idris_file.c

```diff
@@ -223,3 +223,79 @@ FILE *idris2_stdin() { return stdin; }
 FILE *idris2_stdout() { return stdout; }
 
 FILE *idris2_stderr() { return stderr; }
+
+struct child_process {
+  pid_t pid;
+  FILE *in;
+  FILE *out;
+};
+
+// Open a bi-directional pipe, returning the above structure
+// with pid and two file handles.
+struct child_process *idris2_popen2(char *cmd) {
+#ifdef _WIN32
+  return NULL;
+#else
+  pid_t pid;
+  int pipes[4];
+  int err = 0;
+
+  err = pipe(&pipes[0]);
+  if (err) {
+    return NULL;
+  }
+
+  err = pipe(&pipes[2]);
+  if (err) {
+    close(pipes[0]);
+    close(pipes[1]);
+    return NULL;
+  }
+  if ((pid = fork()) > 0) {
```

Comment:
I might need my morning coffee before reading this code, but I think this should be <= 0. Maybe the branches of the conditional got swapped by you at some point?

Reply:
Line 254 is right - the first branch is the parent process which is the one that gets told the new pid (child process gets 0).  The "error checking" on 255-258 I added later, which is unreachable. I'll shift that up above line 254.